uberfire-commons/src/main/java/org/uberfire/commons/cluster/ClusterJMSService.java

```diff
@@ -108,24 +108,32 @@ private ClusterParameters loadParameters() {
                                    Class<T> objectMessageClass,
                                    Consumer<T> listener) {
         try {
-            Session session = createConsumerSession();
-            Destination topic = createDestination(type,
-                                                  channel,
-                                                  session);
-            MessageConsumer messageConsumer = session.createConsumer(topic);
-
-            messageConsumer.setMessageListener(message -> {
-                if (message instanceof ObjectMessage) {
-                    try {
-                        Serializable object = ((ObjectMessage) message).getObject();
-                        if (objectMessageClass.isInstance(object)) {
-                            listener.accept((T) object);
+
+            if (!consumerSessions.containsKey(channel)) {
+                Session session = createConsumerSession();
+                consumerSessions.put(channel, session);
```

Comment:
contains followed by put is not an atomic operation, maybe use computeIfAbsent?

Reply:
+1